packages/smooth_ui_library/lib/animations/smooth_reveal_animation.dart

```diff
@@ -32,19 +32,24 @@ class _SmoothRevealAnimationState extends State<SmoothRevealAnimation>
         vsync: this,
         duration: Duration(milliseconds: widget.animationDuration));
     final CurvedAnimation curve = CurvedAnimation(
-        curve: widget.animationCurve, parent: _animationController);
+        curve: widget.animationCurve, parent: _animationController!);
     _animationOffset =
         Tween<Offset>(begin: widget.startOffset, end: Offset.zero)
             .animate(curve);
 
     Timer(Duration(milliseconds: widget.delay), () {
-      _animationController.forward();
+      if (_animationController != null) {
+        _animationController!.forward();
+      }
     });
   }
 
   @override
   void dispose() {
-    _animationController.dispose();
+    if (_animationController != null) {
+      _animationController!.dispose();
+      _animationController = null;
```

Comment:
I set _animationController to null so that I can tell that it has already been disposed.  I could also set a `_disposed` boolean or something like that if anyone prefers it.

Reply:
I think `_animationController = null` is good.